### PR TITLE
Add gufunc signature to `RandomVariable`'s docstrings

### PR DESCRIPTION
**Pull request by rlouf**
In this PR I add [gufunc](https://numpy.org/doc/stable/reference/c-api/generalized-ufuncs.html#:~:text=Definitions,-Elementary%20Function&text=Each%20ufunc%20consists%20of%20an,different%20parts%20of%20the%20arrays.) signatures to the `RandomVariable`s docstrings. The notation is clear and concise; with knoweldge of broadcasting rules and the definition of `size` the signature makes shapes predictible.

Most distributions behave like `Elemwise` `Op`s, with the exception of:
- Multinomial `(), (n) -> (n)`
- Dirichlet `(k) -> (k)`
- MvNormal `(n), (n,n) -> (n)`
- Categorical `(p) -> ()`
- Choice `(x) -> ()`
- Permutation `(x) -> (x)`

### CODE REVIEW

**Comment by brandonwillard** [comment]
This work is part of https://github.com/aesara-devs/aesara/issues/695.  We need to create a single, consistent interface for this information and apply it to all the (effectively) block-wise `Op`s&mdash;which includes all the scalar `Op`s used by `Elemwise` (i.e. we need to replace all the `nfunc_spec`s).

**Comment by rlouf** [comment]
Regardless of how we implement the interface we need to communicate how the _current_ shape systems works to the users in the documentation, and I think gufunc signatures are the right tool for that.

File: aesara/tensor/random/basic.py [hunk 1]
```diff
@@ -993,6 +1089,11 @@ class HyperGeometricRV(RandomVariable):
     def __call__(self, ngood, nbad, nsample, size=None, **kwargs):
         r"""Draw samples from a geometric distribution.
 
+        Signature
+        ---------
+
+        `() -> ()`
```
Comment (ricardoV94) [inline aesara/tensor/random/basic.py:1095]:
```suggestion
        `(), (), () -> ()`
```

File: aesara/tensor/random/basic.py [hunk 2]
```diff
@@ -1820,6 +2002,11 @@ def _infer_shape(self, size, dist_params, param_shapes=None):
     def __call__(self, x, **kwargs):
         r"""Randomly permute a sequence or a range of values.
 
+        Signature
+        ---------
+
+        `(x) -> ()`
```
Comment (ricardoV94) [inline aesara/tensor/random/basic.py:2008]:
```suggestion
        `(x) -> (x)`
```

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
I think those two were wrong

**Comment by rlouf** [comment]
They were!

**Review by ricardoV94 — APPROVED** [review]

**Comment by rlouf** [comment]
Squashed the commits.

**Comment by rlouf** [comment]
Should we merge this? This documents the current behavior of `RandomVariable`s and as such is useful even before #695 is merged.

File: doc/library/tensor/random/basic.rst [hunk 3]
```diff
@@ -47,7 +47,7 @@ Reference
 Distributions
 ==============
 
-Aesara can produce :class:`RandomVariable`\s that draw samples from many different statistical distributions, using the following :class:`Op`\s.
+Aesara can produce :class:`RandomVariable`\s that draw samples from many different statistical distributions, using the following :class:`Op`\s. :class:`RandomVariable`\s are what NumPy calls *Generalized Universal Functions*: each :class:`Op` carries information that states what the "core" dimensionality of the input is, as well as the corresponding dimensionality of the outputs. The list of core dimensions for each parameter of the random variable is called its *signature*.
```
Comment (brandonwillard) [inline doc/library/tensor/random/basic.rst:50]: This is a bit misleading, because `RandomVariable`s aren't technically `gufunc`s in Aesara (or NumPy?)

**Review by brandonwillard — CHANGES_REQUESTED** [review]
Given how many of these are scalars and, as a result, have trivial signatures, it's hard to say how useful those particular cases are.  

Regardless, it doesn't hurt to add them, but we need to do something about the `gufunc`/`RandomVariable` comment before that.

File: doc/library/tensor/random/basic.rst [hunk 4]
```diff
@@ -47,7 +47,7 @@ Reference
 Distributions
 ==============
 
-Aesara can produce :class:`RandomVariable`\s that draw samples from many different statistical distributions, using the following :class:`Op`\s.
+Aesara can produce :class:`RandomVariable`\s that draw samples from many different statistical distributions, using the following :class:`Op`\s. The :class:`RandomVariable`\s behave similarly to NumPy's *Generalized Universal Functions* (or `gunfunc`): it supports "core" random variable :class:`Op`\s that map distinctly shaped inputs to potentially non-scalar outputs. We document this behavior in the following with `gufunc`-like signatures.
```
Comment (rlouf) [inline doc/library/tensor/random/basic.rst:50]: Is that less confusing @brandonwillard ?

Comment (brandonwillard) [reply doc/library/tensor/random/basic.rst:50]: Yeah, that works.

**Review by brandonwillard — DISMISSED** [review]

**Pull request merged**